src/processor/fingerprint.cpp

```diff
@@ -259,14 +369,127 @@ void key_hash_field::operator()(string_buffer &output)
 
     sha256_hash hasher;
     std::string normalized;
+    // By reserving the largest possible size, it should reduce reallocations
+    // We also add +1 to account for the trailing comma
+    normalized.reserve(max_string_size + 1);
     for (unsigned i = 0; i < keys.size(); ++i) {
-        normalize_string(keys[i], normalized, (i + 1) < keys.size());
+        bool trailing_comma = ((i + 1) < keys.size());
+        normalize_key(keys[i], normalized, trailing_comma);
         hasher << normalized;
     }
 
     hasher.write_digest(output.subspan<8>());
 }
 
+void vector_hash_field::operator()(string_buffer &output)
+{
+    if (value.empty()) {
+        return;
+    }
+
+    sha256_hash hasher;
+    for (unsigned i = 0; i < value.size(); ++i) {
+        hasher << value[i];
+        if ((i + 1) < value.size()) {
+            hasher << ",";
+        }
+    }
+    hasher.write_digest(output.subspan<8>());
+}
+
+void kv_hash_fields::operator()(string_buffer &output)
+{
+    if (value.nbEntries == 0) {
```

Comment:
shouldn't you check that it's indeed a map?

Reply:
True, let me add that check